### PR TITLE
Automated cherry pick of #115702: kubelet: output log even file is rotated

### PR DESCRIPTION
**Pull request by kannon92**
Cherry pick of #115702 on release-1.28.

#115702: kubelet: output log even file is rotated

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note

```

### CODE REVIEW

File: pkg/kubelet/kuberuntime/logs/logs.go [hunk 1]
```diff
@@ -318,6 +318,8 @@ func ReadLogs(ctx context.Context, path, containerID string, opts *LogOptions, r
 	found := true
 	writer := newLogWriter(stdout, stderr, opts)
 	msg := &logMessage{}
+	baseName := filepath.Base(path)
+	dir := filepath.Dir(path)
```
Comment (bart0sh) [inline pkg/kubelet/kuberuntime/logs/logs.go:322]: nit: 
```suggestion
	baseName, dir := filepath.Split(path)
```

Comment (kannon92) [reply pkg/kubelet/kuberuntime/logs/logs.go:322]: I don't think we usually edit cherry-pick PRs.

Comment (bart0sh) [reply pkg/kubelet/kuberuntime/logs/logs.go:322]: /lgtm
then

**Comment by liggitt** [comment]
Unless this is fixing a regression, I would not expect changes like this to be backported.

**Comment by kannon92** [comment]
@liggitt Are your comments blocking? Or should I take this to sig-node?

**Comment by kannon92** [comment]
We felt that this was an important enough bug in our product to push down to previous releases. If upstream does not feel that way, we can continue to patch our product but I thought it was worth a cherry-pick.

**Comment by liggitt** [comment]
> @liggitt Are your comments blocking? Or should I take this to sig-node?

I don't see the case for backport of a bugfix for a long-standing bug that isn't a regression... I'd lean toward keeping it just at HEAD releases. If sig-node leads think this should be an exception to the backport policy, it would be helpful for them to articulate why.

**Comment by kannon92** [comment]
Okay. I'll add this to the agenda and we will see.

**Comment by kannon92** [comment]
Okay. @dchen1107 clarified the policy.

This is not a regression or a critical/security fix so it will not be backported to upstream.

/close

**Pull request closed without merging**